src/librustc_resolve/lib.rs

```diff
@@ -3577,7 +3576,7 @@ impl<'a> Resolver<'a> {
 
         let name = path[path.len() - 1].node.name;
         // Make sure error reporting is deterministic.
-        names.sort_by_key(|name| name.as_str());
+        names.sort_by_cached_key(|name| name.as_str());
```

Comment:
This method is coming from: https://github.com/rust-lang/rust/blob/e38eca6caca2bcb7da36d3075b390625b717fbab/src/libsyntax_pos/symbol.rs#L113-L119, which seemed non-trivial (but perhaps it can be optimised quite effectively?).

Reply:
Hmm, it's probably fairly cheap, but might be worth caching it anyway so the sort code is easier for the compiler to deal with.  Thanks for resolving the type :)